### ISSUE TITLE
Display "ESLint" and the triggered rule within the tooltip

### ISSUE THREAD

**Issue by LittleBrainz** [Microsoft/vscode-eslint#10]
Overall, I am very pleased with vscode-eslint. However, when ESLint inappropriately (from my point of view) flags some offending syntax, it is difficult to determine which rule caused the error.

Also, when several linters are active (including the built VS Code IntelliSense), I am not certain which is causing the text to be highlighted.

Would it be possible to format the tooltip along the lines of:

```
ESLint: you have done something stupid (stupid-syntax-rule)
```

The same format would be ideal if included in the VS Code error summary (ctrl-p + !).

Cheers!

**Comment by egamma** [comment]
Now that I've started to work with TSLint I'm starting to feel the same, that is I'd like to see whether are warning comes from TypeScript or TSLint. 

@dbaeumer I suggest we come up with a guideline and then apply it across all the linters.

My preference would be to show the linter/source at the end of the message, i.e.,

`You have done something stupid (stupid-syntax-rule) [ESLint]`

The message is the 'main' thing and it should come first before the linter which is secondary information.

**Comment by LittleBrainz** [comment]
@egamma - I agree.

**Comment by dbaeumer** [comment]
I think we should not mangle this (from an API perspective) into the message. Instead we should enhance the Diagnostic API with a source field and allow linters to fill in that field. This would allow for proper filtering and grouping in the UI side. See https://github.com/Microsoft/vscode/issues/1010 for the discussion on the VS Code side.

**Comment by egamma** [comment]
I was only taking about the rendering.. Fully agree, that this should not be done with string concatenation, but an API.

**Comment by dbaeumer** [comment]
This got addressed a long time ago.

**Comment by dbaeumer** [comment]
Got fixed a long time ago.
